pallets/torus0/src/namespace.rs

```diff
@@ -74,10 +90,22 @@ impl<T: Config> NamespacePricingConfig<T> {
         Ok(base_fee.saturating_mul(multiplier).into_inner())
     }
 
-    /// Calculates the deposit needed to register a namespace.
+    /// Calculates the deposit needed to register a namespace. If a path with the agent prefix
+    /// is provided and contains more segments, the prefix (agent literal and agent name)
+    /// will be dropped.
     pub fn namespace_deposit(&self, path: &NamespacePath) -> BalanceOf<T> {
+        let segments: Vec<_> = path.segments().collect();
+        let mut segments = &segments[..];
+
+        if segments.first() == Some(&&b"agent"[..]) && segments.len() > 2 {
+            segments = segments.get(2..).unwrap_or(segments);
+        }
+
+        let bytes: u32 = segments.iter().map(|segment| segment.len() as u32).sum();
+        let dots = segments.len().saturating_sub(1) as u32;
+
         self.deposit_per_byte
-            .saturating_mul((path.as_bytes().len() as u32).into())
+            .saturating_mul(bytes.saturating_add(dots).into())
     }
```

Comment:
I'd prefer this not to drop the prefix on the fee computation. In the end, the prefix just adds a simple flat fee (that actually reflects the cost of the storage), and it's simpler, as we don't need to add extra logic.